src/data-type.ts

```diff
@@ -69,14 +69,16 @@ export interface DataType {
 
   declaration(parameter: Parameter) : string,
   writeTypeInfo(buf: any, data: ParameterData, options: InternalConnectionOptions) : void,
-  writeParameterData(buf: any, data: ParameterData, options: InternalConnectionOptions, callback: () => void) : void,
+  writeParameterData(buf: any, data: ParameterData, options: InternalConnectionOptions, callback: () => void): void,
   validate(value: any) : any, // TODO: Refactor 'any' and replace with more specific type.
+  generate(parameter: ParameterData, _options: InternalConnectionOptions): any,
```

Comment:
The proper type for this function would be:

```suggestion
  generate(parameter: ParameterData, _options: InternalConnectionOptions): Generator<Buffer, void>,
```